api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -392,65 +392,87 @@ public static void checkRequiredVersion(String version, String versionRange) thr
 	}
 	
 	/**
-	 * Compares <code>version</code> to <code>value</code> version and value are strings like
-	 * 1.9.2.0 Returns <code>0</code> if either <code>version</code> or <code>value</code> is null.
+	 * Compare two version strings.
 	 *
-	 * @param version String like 1.9.2.0
-	 * @param value String like 1.9.2.0
-	 * @return the value <code>0</code> if <code>version</code> is equal to the argument
-	 *         <code>value</code>; a value less than <code>0</code> if <code>version</code> is
-	 *         numerically less than the argument <code>value</code>; and a value greater than
-	 *         <code>0</code> if <code>version</code> is numerically greater than the argument
-	 *         <code>value</code>
-	 * <strong>Should</strong> correctly comparing two version numbers
-	 * <strong>Should</strong> treat SNAPSHOT as earliest version
+	 * @param versionA String like 1.9.2.0, may include a qualifier like "-SNAPSHOT", may be null
+	 * @param versionB String like 1.9.2.0, may include a qualifier like "-SNAPSHOT", may be null
+	 * @return the value <code>0</code> if versions are equal; a value less than <code>0</code> if first version is
+	 * 		   before the second one; a value greater than <code>0</code> if first version is after the second one.
+	 * 		   If version numbers are equal and only one of them has a qualifier, the version without the qualifier is
+	 * 		   considered greater.
 	 */
-	public static int compareVersion(String version, String value) {
+	public static int compareVersion(String versionA, String versionB) {
+		return compareVersion(versionA, versionB, false);
+	}
+
+	/**
+	 * Compare two version strings. Any version qualifiers are ignored in the comparison.
+	 *
+	 * @param versionA String like 1.9.2.0, may include a qualifier like "-SNAPSHOT", may be null
+	 * @param versionB String like 1.9.2.0, may include a qualifier like "-SNAPSHOT", may be null
+	 * @return the value <code>0</code> if versions are equal; a value less than <code>0</code> if first version is
+	 * 		   before the second one; a value greater than <code>0</code> if first version is after the second one.
+	 */
+	public static int compareVersionIgnoringSnapshotQualifier(String versionA, String versionB) {
```

Comment:
Here `SnapshotQualifier` is misleading because the qualifier may be something else than "SNAPSHOT". I will rename this function.